thewalrus/reference.py

```diff
@@ -375,3 +376,51 @@ def rspm(s):
     """
     gen = rpmp(s)
     return chain(*(splitter(i) for i in gen))
+
+
+def mtl(A):
+    """Returns the Montrealer of an NxN matrix and an N-length vector.
+
+    Args:
+        A (array): an NxN array of even dimensions. Can be symbolic.
+
+    Returns:
+        np.float64, np.complex128 or sympy.core.add.Add: the Montrealer of matrix A.
+    """
+    n,_ = A.shape
+    net_sum = 0
+
+    for s in rpmp(range(n)):
+        net_prod = 1
+        for a in s:
+            a = sorted(a)
+            net_prod *= A[a[0],a[1]]
+            
+        net_sum += net_prod
+
+    return net_sum
+
+
+def lmtl(A, zeta):
+    """Returns the loop Montrealer of an NxN matrix and an N-length vector.
+
+    Args:
+        A (array): an NxN array of even dimensions. Can be symbolic.
+        zeta (array): an N-length vector of even dimensions. Can be symbolic.
+
+    Returns:
+        np.float64, np.complex128 or sympy.core.add.Add: the loop Montrealer of matrix A, vector zeta
+    """
+
+    fill_diagonal(A, zeta.conj())
+    n,_ = A.shape
+    net_sum = 0
+
+    for s in rspm(range(n)):
+        net_prod = 1
+        for a in s:
+            net_prod *= A[a[0],a[1]]
```

Comment:
Here you could check if `a[0]==a[1]` then use `zeta[a[0]]`, that way the function in `reference` have minimal dependencies, which is something we very much want.